cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -121,38 +121,46 @@ func createPoolBuilder(cStorPool *apis.CStorPool, diskList []string) []string {
 	createAttr = append(createAttr, poolNameUID)
 	poolType := cStorPool.Spec.PoolSpec.PoolType
 	if poolType == "striped" {
-		for _, disk := range diskList {
-			createAttr = append(createAttr, disk)
-		}
+		createAttr = append(createAttr, blockDeviceList...)
 		return createAttr
 	}
 	// To generate pool of the following types:
-	// mirrored (grouped by multiples of 2): mirror disk1 disk2 mirror disk3 disk4
-	// raidz (grouped by multiples of 3): raidz disk1 disk2 disk3 raidz disk 4 disk5 disk6
-	// raidz2 (grouped by multiples of 6): raidz2 disk1 disk2 disk3 disk4 disk5 disk6
-	for i, disk := range diskList {
+	// mirrored (grouped by multiples of 2): mirror blockdevice1 blockdevice2 mirror blockdevice3 blockdevice4
+	// raidz (grouped by multiples of 3): raidz blockdevice1 blockdevice2 blockdevice3 raidz blockdevice 4 blockdevice5 blockdevice6
+	// raidz2 (grouped by multiples of 6): raidz2 blockdevice1 blockdevice2 blockdevice3 blockdevice4 blockdevice5 blockdevice6
+	for i, bd := range blockDeviceList {
 		if i%defaultGroupSize[poolType] == 0 {
 			createAttr = append(createAttr, poolTypeCommand[poolType])
 		}
-		createAttr = append(createAttr, disk)
+		createAttr = append(createAttr, bd)
 	}
 
 	return createAttr
 }
 
-// CheckValidPool checks for validity of CStorPool resource.
-func CheckValidPool(cStorPool *apis.CStorPool, devID []string) error {
+// ValidatePool checks for validity of CStorPool resource.
+func ValidatePool(cStorPool *apis.CStorPool, devID []string) error {
 	poolUID := cStorPool.ObjectMeta.UID
 	if len(poolUID) == 0 {
 		return fmt.Errorf("Poolname/UID cannot be empty")
 	}
 	diskCount := len(devID)
 	poolType := cStorPool.Spec.PoolSpec.PoolType
 	if diskCount < defaultGroupSize[poolType] {
-		return errors.Errorf("Expected %v no of disks, got %v no of disks for pool type: %v", defaultGroupSize[poolType], diskCount, poolType)
+		return errors.Errorf(
+			"validate pool failed: expected {%d} blockdevices got {%d}, for pool type {%s}",
```

Comment:
n_p: "csp validation failed: ..." imparts more clarity to me.

Reply:
Done